app/screens/home/recent_mentions/index.tsx

```diff
@@ -1,24 +1,110 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {useRoute} from '@react-navigation/native';
-import React, {useCallback, useState} from 'react';
-import {Text} from 'react-native';
+import {Q} from '@nozbe/watermelondb';
+import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
+import withObservables from '@nozbe/with-observables';
+import {useIsFocused, useNavigation} from '@react-navigation/native';
+import compose from 'lodash/fp/compose';
+import groupBy from 'lodash/groupBy';
+import moment from 'moment-timezone';
+import React, {useCallback, useState, useEffect, useMemo} from 'react';
+import {useIntl} from 'react-intl';
+import {StyleSheet, View, ActivityIndicator, SectionList, SectionListRenderItemInfo} from 'react-native';
 import Animated, {useAnimatedStyle, withTiming} from 'react-native-reanimated';
 import {SafeAreaView} from 'react-native-safe-area-context';
+import {of as of$, map as map$} from 'rxjs';
+import {catchError, switchMap} from 'rxjs/operators';
 
-const RecentMentionsScreen = () => {
-    const route = useRoute();
-    const params = route.params! as {direction: string};
-    const toLeft = params.direction === 'left';
-    const [start, setStart] = useState(false);
+import {getRecentMentions} from '@actions/remote/search';
+import DateSeparator from '@app/components/post_list/date_separator';
+import {MM_TABLES, SYSTEM_IDENTIFIERS} from '@app/constants/database';
+import {SystemModel, UserModel} from '@app/database/models/server';
+import NavigationHeader from '@components/navigation_header';
+import {LARGE_HEADER_PROGRESSVIEW_OFFSET} from '@constants/view';
+import {useServerUrl} from '@context/server_url';
+import {withTheme} from '@context/theme';
+import {useCollapsibleHeader} from '@hooks/header';
 
-    const onLayout = useCallback(() => {
-        setStart(true);
-    }, []);
+import Mention from './components/mention';
+
+import type {WithDatabaseArgs} from '@typings/database/database';
+import type PostModel from '@typings/database/models/servers/post';
+
+const AnimatedSectionList = Animated.createAnimatedComponent(SectionList);
+
+const {USER, SYSTEM, POST} = MM_TABLES.SERVER;
+
+type Props = {
+    mentions: PostModel[];
+    theme: Theme;
+    currentUser: UserModel;
+}
+
+type Section = {
+    key: string;
+    title: string;
+    data: PostModel[];
+    keyExtractor: (item: PostModel) => string;
+}
+
+const styles = StyleSheet.create({
+    loading: {
+        alignItems: 'center',
+        flex: 1,
+        justifyContent: 'center',
+    },
+});
+
+function getSections(items: PostModel[]) {
```

Comment:
it doesn't look like you need a section list, the date separator can be part of the FlatList, reason: no really difference between sections data, and FlatList is a bit better around perf than SectionList plus reducing the an iteration to group the results, you could just sort them and iterate on it just once

Reply:
done